otus/freebsd/src/sys/dev/athp/if_athp_core.c

```diff
@@ -1690,18 +1690,41 @@ ath10k_core_probe_fw(struct ath10k *ar)
 	return ret;
 }
 
+static void
+clean_ath10k_core_probe_fw(struct ath10k * ar) {
+	//This function has been made to do cleanup to prevent memory leaks if any exist.
+	//First step tell bmi done has not been sent so it will re-setup bmi.
+	ar->bmi.done_sent = false;
+}
+
+static int
+attempt_ath10k_core_probe_fw(struct ath10k *ar, int anum) {
+	//anum is the number of attempts that have been tried.
+	//This is to reload the firmware multiple times because sometimes it fails for no reason,
+	//it may be a freebsd only issue.
+	int status = ath10k_core_probe_fw(ar);
+	if (status) {
+		ath10k_err(ar, "could not probe fw, clean up allocations and memory and retry. (%d)\n", status);
+		pause_sig("pausing to wait for the ath cpu to be ready.", 1000);
```

Comment:
pause_sig isn't used elsewhere and it doesn't grab a lock; can you see if there's a different pause/tsleep that can take a suitable lock?

Reply:
there is a pause and tsleep I can use that uses a lock, what lock would you want me to use, and sleep on?